### PR TITLE
doc: Add documentation for power optimization user guide

### PR DESCRIPTION
**Pull request by umapraseeda**
Add documentation for power optimization user guide

Signed-off-by: Uma Praseeda <uma.praseeda@nordicsemi.no>
[Power optimization — nRF Connect SDK 1.4.99 documentation.pdf](https://github.com/nrfconnect/sdk-nrf/files/5588501/Power.optimization.nRF.Connect.SDK.1.4.99.documentation.pdf)

### CODE REVIEW

**Comment by umapraseeda** [comment]
@joakimtoe

**Comment by umapraseeda** [comment]
[Power optimization — nRF Connect SDK 1.4.99 documentation.pdf](https://github.com/nrfconnect/sdk-nrf/files/5597941/Power.optimization.nRF.Connect.SDK.1.4.99.documentation.pdf)

File: doc/nrf/app_power_opt.rst [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+For a similar configuration, see the :ref:`udp` sample, which is optimized for low power operation on the ``nRF9160dk_nrf9160ns`` build target without any modifications.
+
+
+Idle current due to other peripherals
+=====================================
+
+Elevated currents can be caused by peripherals other than the serial ports.
+
+The power management of the Nordic SoCs automatically switches in and out the resources that are needed by the active peripherals.
+Peripherals that need a high frequency clock like UART, PWM and PDM, will show similar elevated currents if the high frequency timers associated with these peripherals are not turned off during sleep.
```
Comment (jtguggedal) [inline doc/nrf/app_power_opt.rst:87]: `high frequency clock` and `high frequency timer` should be aligned.
I was a little confused by this and the previous sentence. The first states that the power management automatically enables and disables resource required by the peripherals, while the latter says that the users must control the clocks required by peripherals, or am I reading the last sentence wrong?

File: doc/nrf/app_power_opt.rst [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+
+
+Simulation using Online Power Profiler
+--------------------------------------
+
+To simulate the use case with Online Power Profiler, complete the following steps:
+
+1. Start the simulation in Online Power Profiler with the default settings and add the following changes to the default configuration:
+
+   A. In the :guilabel:`PSM` tab on the left, set Active time(sec) to ``0`` (indicating that the device need not be active after sending the payload).
```
Comment (jtguggedal) [inline doc/nrf/app_power_opt.rst:135]: Missing space `time(sec)`

File: doc/nrf/app_power_opt.rst [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+
+
+   If PSM is activated, you can see that the current value drops to a few µA.
+   As long as the PSM mode is activated and no other activity occurs, the current value remains in the PSM floor as shown in the above image until a transmission occurs or until the PSM timer times out. Even though the requested time out value in the use case is 60 minutes, it is decided by the network.
+   The PSM floor current is 4.69 µA as seen in the above image.
+
+#. Since the data transfer interval is 20 minutes, you can increase the frequency of transmission. Set :option:`CONFIG_UDP_DATA_UPLOAD_FREQUENCY_SECONDS` to ``120`` in :file:`prj.conf`.
+#. Re-build and program the sample.
+#. Observe that a 40-byte data packet is sent every two minutes. Make sure that the frequency is sufficiently higher than the duration of RRC inactivity present in the network. RRC inactivity timer value can be over one minute in some networks, and to get correct measurements, you must ensure that the device has gone into PSM before the new data packet is sent. After a duration of two minutes, observe the following results in the Power Profiler Kit II interface:
```
Comment (jtguggedal) [inline doc/nrf/app_power_opt.rst:192]: This is the first time `RRC inactivity` is mentioned, and it's probably a new term to many users. Might point to an explanation or include it somewhere in this document

File: doc/nrf/app_power_opt.rst [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+1. Measure the RRC inactivity timer value using Power Profiler Kit II.
+
+
+   Below image shows how the RRC inactivity timer can be measured with Power Profiler Kit II:
+
+   .. figure:: images/app_power_opt_ppk_inactivity_timer.png
+      :scale: 50 %
+      :alt: Power Profiler Kit II RRC inactivity timer measurement
+
+   For the use case, the RRC inactivity timer is 11 seconds as shown in the above image.
```
Comment (jtguggedal) [inline doc/nrf/app_power_opt.rst:252]: `the use case` -> `this use case`?

File: doc/nrf/app_power_opt.rst (hunk 1 above)
Comment (jhn-nordic) [reply doc/nrf/app_power_opt.rst:87]: What about replacing the last sentence with: "Peripherals that need a high frequency clock like i.e UART, PWM, PDM or high frequency timers, will give similar currents when enabled"

File: doc/nrf/app_power_opt.rst [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+
+Power profiling assumes importance in the case of elevated current consumption, which occurs typically in the following scenarios:
+
+* Serial logging
+* Idle current due to other peripherals
+
+Serial logging
+==============
+
+Current measurements on devices that have the |NCS| samples or applications programmed with the default configuration, might show elevated current values.
```
[the remark this answers is not in the document]
Comment (jhn-nordic) [reply doc/nrf/app_power_opt.rst:34]: elevated currents compared to what you would expect from Nordics ultra low power SoCs. (was what it was compared to)

**Review by jtguggedal — APPROVED** [review]

**Review by jhn-nordic — APPROVED** [review]

**Comment by jhn-nordic** [comment]
Looks good now!

**Comment by ioannisg** [comment]
@umapraseeda  is the DNM label still applicable?

**Comment by umapraseeda** [comment]
> @umapraseeda is the DNM label still applicable?

This is ready to merge now.

**Pull request merged**